app/assets/javascripts/discourse/tests/integration/components/widgets/render-glimmer-test.js

```diff
@@ -231,6 +247,39 @@ module("Integration | Component | Widget | render-glimmer", function (hooks) {
     assert.true(DemoWidget.actionTriggered, "widget event is triggered");
   });
 
+  test("modify widget state with component action", async function (assert) {
+    await render(
+      hbs`<MountWidget @widget="demo-widget" @args={{hash arg1="val1"}} />`
+    );
+
+    assert.false(
+      DemoWidget.actionTriggered,
+      "widget event has not been triggered yet"
+    );
+
+    assert.strictEqual(
+      query(".action-state").innerText,
+      "false",
+      "eventTriggered is false in nested component"
+    );
+
+    assert.true(
+      exists("div.demo-component button"),
+      "component button is rendered"
+    );
+
+    await click("div.demo-component button");
+    assert.true(DemoWidget.actionTriggered, "widget event is triggered");
+
+    await settled();
```

Comment:
This shouldn't be required, since `click` helper already await for the settled state